main.go

```diff
@@ -180,3 +204,19 @@ func stdHandler(ctx context.Context, okCh chan<- int, errCh chan<- error) io.Wri
 
 	return writer
 }
+
+// getFreeMongoPort is simple utility to find a free port on the "localhost" interface of the host machine
+// for a local mongo server to use. If any error occurs the default mongo port of "27017" is returned
```

Comment:
What if 27017 is used? Could we make this function return an error and then deal it before trying to create the server?

Reply:
Yeah good question about returning the 27017, and although I did't really want the function to return an error, looking at the concern below, I think I'll change it to return an error